### PR TITLE
access_logs: fix overenthusiastic DPE

### LINKED ISSUE #13274: http2: regression in DPE accounting when remote resets
As of https://github.com/envoyproxy/envoy/pull/13052, and this diff specifically: 

https://github.com/envoyproxy/envoy/pull/13052/files#diff-ed138bc8d20eee664368960ecfa29a73R930-R936

We are now setting response details for remote resets.

This causes this code to kick in:

https://github.com/envoyproxy/envoy/blob/c71ec2729cc3c0708223d303e0f24e3bf9a5d0eb/source/common/http/conn_manager_impl.cc#L1401-L1407

Which then logs these resets as DPE which I think is incorrect. We need to be more specific in terms of which response details cause DPE accounting.

### PR DESCRIPTION
**Pull request by alyssawilk**
Commit Message: Fixing a bug where stream resets got flagged as downstream protocol errors
Risk Level: low
Testing: integration tested
Docs Changes: n/a
Release Notes: n/a
Fixes #13274

### CODE REVIEW

File: source/common/http/conn_manager_impl.cc [hunk 1]
```diff
@@ -1402,7 +1403,10 @@ void ConnectionManagerImpl::ActiveStream::onResetStream(StreamResetReason, absl:
   // DownstreamProtocolError and propagate the details upwards.
   const absl::string_view encoder_details = response_encoder_->getStream().responseDetails();
   if (!encoder_details.empty()) {
-    filter_manager_.streamInfo().setResponseFlag(StreamInfo::ResponseFlag::DownstreamProtocolError);
+    if (reset_reason == StreamResetReason::LocalReset) {
```
Comment (snowp) [inline source/common/http/conn_manager_impl.cc:1406]: Maybe collapse this with the previous if block to reduce nesting? Just && the two conditions

**Review by snowp — CHANGES_REQUESTED** [review]
Thanks! Just one comment

**Comment by alyssawilk** [comment]
if collapsed, macos failure looks unrelated.

**Review by snowp — APPROVED** [review]
Thanks!

**Pull request merged**